signing_service/signing_service/tests/tests.py

```diff
@@ -116,6 +139,67 @@ def test_that_signing_service_should_reraise_different_socket_erros(self):
         mock_socket_close.assert_called_once()
 
 
+class SigningServiceHandleConnectionTestCase(TestCase):
+
+    def setUp(self):
+        self.host = '127.0.0.1'
+        self.port = 8000
+        self.initial_reconnect_delay = 2
+
+    def test_that__handle_connection_should_send_wrapped_service_refuse_if_frame_is_invalid(self):
+        # Prepare message with wrong signature
+        middleman_message = GolemMessageFrame(Ping(), 99)
+        raw_message = append_frame_separator(
+            escape_encode_raw_message(
+                middleman_message.serialize(
+                    private_key=CONCENT_PRIVATE_KEY,
+                )
+            )
+        )
+        malformed_raw_message = bytes(bytearray([max(raw_message[1] - 1, 0)])) + raw_message[1:]
+
+        def mocked_generator(connection):  # pylint: disable=unused-argument
+            yield malformed_raw_message
+            raise SigningServiceValidationError()
+
+        with mock.patch('signing_service.signing_service.SigningService.run'):
+            with mock.patch(
+                'signing_service.signing_service.unescape_stream',
+                side_effect=mocked_generator,
+            ):
+                with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as signing_service_socket:
+                    signing_service_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as client_socket:
+                        client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                        signing_service = SigningService(
+                            self.host,
+                            self.port,
+                            self.initial_reconnect_delay,
+                            CONCENT_PUBLIC_KEY,
+                            SIGNING_SERVICE_PRIVATE_KEY,
+                        )
+
+                        # For test purposes we reverse roles, so signing service works as server.
+                        signing_service_socket.bind(('127.0.0.1', 8001))
```

Comment:
this is a bit risky... please use `unused_tcp_port` fixture (https://github.com/pytest-dev/pytest-asyncio#unused_tcp_port)